packages/app/src/cli/services/dev/update-extension.ts

```diff
@@ -83,23 +87,29 @@ export async function updateExtensionConfig({
   }
 
   let configObject = await loadConfigurationFile(extension.configuration.path)
-  const {extensions} = ExtensionsArraySchema.parse(configObject)
 
-  if (extensions) {
-    // If the config has an array, find our extension using the handle.
-    const configuration = await parseConfigurationFile(UnifiedSchema, extension.configuration.path, abort)
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const extensionConfig = configuration.extensions.find((config: any) => config.handle === extension.handle)
-    if (!extensionConfig) {
-      abort(
-        `ERROR: Invalid handle
-  - Expected handle: "${extension.handle}"
-  - Configuration file path: ${relativizePath(extension.configuration.path)}.
-  - Handles are immutable, you can't change them once they are set.`,
-      )
-    }
+  // Get configuration object for app config modules and skip handle check
+  if (extension.specification.appModuleFeatures().includes('app_config')) {
```

Comment:
TODO: I think `appModuleFeatures().includes('app_config')` is repeated a bunch, and we may already have a helper to DRY this up.